openmc/statepoint.py

```diff
@@ -92,13 +91,8 @@ class StatePoint(object):
     """
 
     def __init__(self, filename):
-        if filename.endswith('.h5'):
-            import h5py
-            self._f = h5py.File(filename, 'r')
-            self._hdf5 = True
-        else:
-            self._f = open(filename, 'rb')
-            self._hdf5 = False
+        import h5py
+        self._f = h5py.File(filename, 'r')
```

Comment:
I think for the Python API and scripts, there should be no expectation that the file extension is anything in particular. If the file is not an HDF5 file, the user will get an IOError telling them that the 'file signature is not found' which is the proper behavior as I see it. If you feel that error message is too cryptic, we could catch that exception and re-raise it with a more intuitive message.